app/services/process_stripe_payment.rb

```diff
@@ -0,0 +1,25 @@
+class ProcessStripePayment
+  attr_reader :controller, :amount, :email, :token
+  
+  def initialize(controller, amount, email, token)
+    @controller = controller
+    @amount = amount
+    @email = email
+    @token = token
+  end
+  
+  def charge_card
+    begin
+      Stripe.api_key = ENV['STRIPE_SECRET_KEY']
+      Stripe::Charge.create(
+        amount: amount,
+        currency: "gbp",
+        source: token,
+        description: "Charge for #{email}"
+      ) 
+    rescue Stripe::CardError => e
+      controller.flash[:danger] = e.message
```

Comment:
I just found another way of doing this and it means I don't have to pass the controller or display the flash message in the service object.  The issue I had was that I had to return false with a payment error and then would have to pass the message. I didn't realise I could return multiple value.  Again this probably isn't the ideal solution but at least it sorts some of the issues for now.